### PR TITLE
Add functionality to update long url

### PR DESCRIPTION
**Pull request by gaurav-solanki97**

### CODE REVIEW

File: app/Http/routes.php [hunk 1]
```diff
@@ -67,6 +67,7 @@
     $app->post('action/shorten', ['as' => 'api_shorten_url', 'uses' => 'ApiLinkController@shortenLink']);
     $app->get('action/shorten', ['as' => 'api_shorten_url', 'uses' => 'ApiLinkController@shortenLink']);
     $app->post('action/shorten_bulk', ['as' => 'api_shorten_url_bulk', 'uses' => 'ApiLinkController@shortenLinksBulk']);
+    $app->post('action/edit_link_long_url', ['as' => 'api_edit_link_long_url', 'uses' => 'ApiLinkController@editLinkLongUrl']);
```
Comment (sagar-vaghela97) [inline app/Http/routes.php]:
```suggestion
    $app->put('action/shorten', ['as' => 'api_edit_link_long_url', 'uses' => 'ApiLinkController@editLinkLongUrl']);
```

Comment (gaurav-solanki97) [reply app/Http/routes.php]: Okay

Comment (gaurav-solanki97) [reply app/Http/routes.php]: Done.

File: app/Http/routes.php [hunk 2]
```diff
@@ -67,6 +67,7 @@
     $app->post('action/shorten', ['as' => 'api_shorten_url', 'uses' => 'ApiLinkController@shortenLink']);
     $app->get('action/shorten', ['as' => 'api_shorten_url', 'uses' => 'ApiLinkController@shortenLink']);
     $app->post('action/shorten_bulk', ['as' => 'api_shorten_url_bulk', 'uses' => 'ApiLinkController@shortenLinksBulk']);
+    $app->put('action/shorten', ['as' => 'api_edit_link_long_url', 'uses' => 'ApiLinkController@editLinkLongUrl']);
```
Comment (sagar-vaghela97) [inline app/Http/routes.php]:
```suggestion
    $app->put('action/shorten/short_url/{url}', ['as' => 'api_edit_link_long_url', 'uses' => 'ApiLinkController@editLinkLongUrl']);
```

File: app/Http/Controllers/Api/ApiLinkController.php [hunk 3]
```diff
@@ -137,4 +137,25 @@ public function lookupLink(Request $request) {
             throw new ApiException('NOT_FOUND', 'Link not found.', 404, $response_type);
         }
     }
+
+    public function editLinkLongUrl(Request $request)
+    {
+        $link_ending = $request->input('link_ending');
```
Comment (sagar-vaghela97) [inline app/Http/Controllers/Api/ApiLinkController.php]:
```suggestion
        $link_ending = $request->input('short-url');
```

File: app/Http/Controllers/Api/ApiLinkController.php [hunk 4]
```diff
@@ -137,4 +137,25 @@ public function lookupLink(Request $request) {
...
+        $link = LinkHelper::linkExists($link_ending);
+
+        $new_long_url = $request->input('new_long_url');
```
Comment (sagar-vaghela97) [inline app/Http/Controllers/Api/ApiLinkController.php]:
```suggestion
        $new_long_url = $request->input('long_url');
```

**Review by sagar-vaghela97 — APPROVED** [review]

**Pull request merged**